src/graph/sampler.cc

```diff
@@ -524,4 +525,198 @@ IdArray SamplerOp::RandomWalk(
   return traces;
 }
 
+void ConstructLayers(const int64_t *indptr,
+                     const dgl_id_t *indices,
+                     const IdArray seed_array,
+                     const std::vector<size_t> &layer_sizes,
+                     std::vector<dgl_id_t> *layer_offsets,
+                     std::vector<dgl_id_t> *node_mapping,
```

Comment:
`node_mapping` is confusing. maybe rename it with `node_set` or `layer_node_set`?

Reply:
`node_mapping` here corresponds to `NodeFlow::node_mapping`.